api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -92,4 +94,117 @@ export class ProjectAclService implements ProjectAccessControl {
       this.canViewSolutionRoles,
     );
   }
+
+  async checkUserIsOwner(userId: string, projectId: string): Promise<Permit> {
+    const userIsProjectOwner = await this.roles.findOne({
+      where: {
+        projectId,
+        userId,
+        roleName: Roles.project_owner,
+      },
+    });
+    if (!userIsProjectOwner) {
+      return false;
+    }
+
+    return true;
+  }
+
+  async checkLastOwner(userId: string, projectId: string): Promise<Permit> {
+    const allOwnersInProject = await this.roles.find({
+      where: {
+        projectId,
+        roleName: Roles.project_owner,
+      },
+    });
+    if (
+      allOwnersInProject.length === 1 &&
+      allOwnersInProject[0].userId === userId
+    ) {
+      return false;
+    }
+    return true;
```

Comment:
I can't really explain why, but I'm not the biggest fan of this... We could add the `userId` condition to the where clause and then just straight away `return allOwnersInProject.length <= 1`

But again, this is mostly a preference of syntax, nothing blocking :sweat_smile:

Reply:
If we add the `userId` condition to the allOwners it would only always find one as long as it exists (thus making impossible to remove other owners from the project).

If I do the same but I don't add the `userId` to the where clause you would have the possibility of removing yourself even if you are the last one in the project.

What I would do (and I'm going to commit is changing what is returned into `return allOwnersInProject.length > 1` so it would only allow to delete owners as long as there is more than 1, and it would allow to remove yourself as owner (that I doubt if it should be possible, but this is for another time)

Reply:
@rubvalave @ikas is there a reason why we cannot return count ?

(pseudocode)
``` 
select count(....) where projectId = ... and role = ... and userId != ....
 ```